### PR TITLE
DDP-5848 provide unit tests for lazy-resource directive changes

### PR DESCRIPTION
**Pull request by aglushko-broad**

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/* tslint:disable:max-classes-per-file */
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { Component, DebugElement } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { By } from '@angular/platform-browser';
+import { LazyLoadResourcesDirective, WindowRef } from 'ddp-sdk';
+import { FormsModule } from '@angular/forms';
+
+fdescribe('LazyLoadResourcesDirective', () => {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: please remove 'f'.
`fdescribe` => `describe`

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
+    fixture.detectChanges();
+  }));
+
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  fit('updates src', () => {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: `fit` => `it`

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          this.callback([{isIntersecting: true} as IntersectionObserverEntry]);
+        };
+        this.unobserve = jasmine.createSpy('unobserve');
+      }
+    }
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    (window as any).IntersectionObserver = IntersectionObserver;
+
+    TestBed.configureTestingModule({
+      imports: [
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: Seems like these imports are not needed.

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        LazyLoadResourcesDirective
+      ],
+      providers: [
+        { provide: WindowRef, useValue: {nativeWindow: {IntersectionObserver: {}}} },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: unused `debugElement`

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts (hunk 1 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: thanks! strange there is no pre-push hook for that. I guess we need to add it at some point

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts (hunk 2 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: fixed

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts (hunk 3 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: fixed

File: ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts (hunk 4 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/directives/lazyLoadResources.directive.spec.ts]: fixed

**Review by alextanlan — APPROVED** [review]

**Pull request merged**